crates/editor/src/element.rs

```diff
@@ -1477,6 +1478,66 @@ impl EditorElement {
                 }
             }
 
+            if layout.is_singleton && scrollbar_settings.diagnostics {
+                let max_point = layout
+                    .position_map
+                    .snapshot
+                    .display_snapshot
+                    .buffer_snapshot
+                    .max_point();
+
+                let mut entries = layout
+                    .position_map
+                    .snapshot
+                    .buffer_snapshot
+                    .diagnostics_in_range::<_, Point>(Point::zero()..max_point, false)
+                    .collect::<Vec<_>>();
+
+                // We want to sort by severity, in order to paint the most severe diagnostics last.
+                entries.sort_by_key(|entry| std::cmp::Reverse(entry.diagnostic.severity));
```

Comment:
NIT: I do not think that it changes things drastically, but may be good to remove one obvious allocation by
* removing that `.collect::<Vec<_>>();` entirely
* using itertools' sorting methods (albeit they will use Vec under the hood, still looks more clean?)
* iterating over the iterator directly in `for entry in entries`
* also the name could be `diagnostics`, not some generic `entry`

Up to you though, it's mostly a styling change.